classes/class-site-size.php

```diff
@@ -163,32 +163,41 @@ public function recursive_filesize_scanner() {
 	}
 
 	/**
-	 * Get the total filesize for a given file or directory
+	 * Get the total filesize for a given file or directory. Aware of exclusions.
 	 *
-	 * If $file is a file then just return the result of `filesize()`.
-	 * If $file is a directory then recursively calculate the size.
+	 * If $file is a file then return the result of `filesize()` or 0 if it's excluded.
+	 * If $file is a directory then recursively calculate the size without
+	 * the size of excluded files/directories.
 	 *
-	 * @param \SplFileInfo   $file The file or directory you want to know the size of
+	 * @param \SplFileInfo   $file The file or directory you want to know the size of.
 	 *
-	 * @return int           The total filesize of the file or directory
+	 * @return int           The total filesize of the file or directory without
+	 *                       the size of excluded files/directories.
 	 */
 	public function filesize( \SplFileInfo $file ) {
 
-		// Skip missing or unreadable files
+		// Skip missing or unreadable files.
 		if ( ! file_exists( $file->getPathname() ) || ! $file->getRealpath() || ! $file->isReadable() ) {
 			return 0;
 		}
 
-		// If it's a file then just pass back the filesize
+		// If it's a file then return its filesize or 0 if it's excluded.
 		if ( $file->isFile() ) {
-			return $file->getSize();
+
+			if (
+				$this->excludes
```

Comment:
I _think_ this will only catch exact exclude matches but not files that are excluded because one of it's parent folders was excluded. Instead you'll need to do a `preg_match` ala https://github.com/humanmade/backupwordpress/blob/master/admin/schedule-form-excludes.php#L236